packages/shared-internals/src/dep-validation.ts

```diff
@@ -0,0 +1,108 @@
+import Package from './package';
+
+// For each package in the graph, discover all the paths to reach that package.
+// That is, we're identifying all its consumers.
+export function crawlDeps(startingPackage: Package): Map<Package, Package[][]> {
+  let queue: { pkg: Package; path: Package[] }[] = [{ pkg: startingPackage, path: [] }];
+  let seen = new Set<Package>();
+  let results = new Map<Package, Package[][]>();
+  for (;;) {
```

Comment:
unrelated to the goal of the Pr, but moreso a curiosity, 
I suspect this is equiv to `while (true)` -- and in any case is used to to iterate over `queue`, but safely as its contents are changing during iteration.

any reason to favor one method of implicitly-bounded looping over another?